code/modules/mob/living/carbon/human/species_types/ethereal.dm

```diff
@@ -89,9 +89,17 @@
 	UnregisterSignal(former_ethereal, COMSIG_LIGHT_EATER_ACT)
 	QDEL_NULL(ethereal_light)
 	if(ishuman(former_ethereal))
-		update_mail_goodies(former_ethereal)
+		new_species.update_mail_goodies(former_ethereal)
 	return ..()
 
+/datum/species/ethereal/update_quirk_mail_goodies(mob/living/carbon/human/recipient, datum/quirk/quirk, list/mail_goodies = list())
+	if(isnull(quirk))
+		return
+	if(quirk.type == /datum/quirk/blooddeficiency)
```

Comment:
```suggestion
	if(istype(quirk, /datum/quirk/blooddeficiency))
```

don't access the type var, use `istype()` instead. `istype()` also nullchecks for us as well too (since null isn't a type of anything). much cleaner implementation.